src/Compilers/VisualBasic/Portable/Symbols/WellKnownMembers.vb

```diff
@@ -362,20 +362,30 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                 Dim mdName As String = WellKnownTypes.GetMetadataName(type)
                 Dim result As NamedTypeSymbol
+                Dim conflicts As (AssemblySymbol, AssemblySymbol) = Nothing
 
                 If IsTypeMissing(type) Then
                     result = Nothing
                 Else
-                    result = Me.Assembly.GetTypeByMetadataName(mdName, includeReferences:=True, isWellKnownType:=True, useCLSCompliantNameArityEncoding:=True,
+                    result = Me.Assembly.GetTypeByMetadataName(mdName, includeReferences:=True, isWellKnownType:=True, useCLSCompliantNameArityEncoding:=True, conflicts:=conflicts,
                                                                ignoreCorLibraryDuplicatedTypes:=Me.Options.IgnoreCorLibraryDuplicatedTypes)
                 End If
 
                 If result Is Nothing Then
                     Dim emittedName As MetadataTypeName = MetadataTypeName.FromFullName(mdName, useCLSCompliantNameArityEncoding:=True)
 
                     If type.IsValueTupleType() Then
-                        result = New MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo(Assembly.Modules(0), emittedName,
-                                       Function(t) ErrorFactory.ErrorInfo(ERRID.ERR_ValueTupleTypeRefResolutionError1, t))
+
+                        Dim delayedErrorInfo As Func(Of MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo, DiagnosticInfo)
+                        If conflicts.Item1 Is Nothing Then
+                            Debug.Assert(conflicts.Item2 Is Nothing)
+
+                            delayedErrorInfo = Function(t) ErrorFactory.ErrorInfo(ERRID.ERR_ValueTupleTypeRefResolutionError1, t)
+                        Else
+                            delayedErrorInfo = Function(t) ErrorFactory.ErrorInfo(ERRID.ERR_ValueTupleResolutionAmbiguous3, t, conflicts.Item1, conflicts.Item2)
```

Comment:
:memo: I didn't find a better way to deal with allocation for capture here.

Reply:
Perhaps we can avoid allocating a closure for other scenarios by declaring an extra local in this ```Else``` block, initializing it with ```conflicts``` and using it in the lambda instead. 

---
In reply to: [146401097](https://github.com/dotnet/roslyn/pull/22793#discussion_r146401097) [](ancestors = 146401097)

Reply:
It looks like creating a local in the `Else` doesn't make a difference, but the baseline doesn't involve a closure after all. I'll leave the PR as-is.

From this PR:
```
bool flag6 = conflicts.Item1 == null;
Func<MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo, DiagnosticInfo> delayedErrorInfo;
if (flag6)
{
	Debug.Assert(conflicts.Item2 == null);
	delayedErrorInfo = ((VisualBasicCompilation._Closure$__.$I257-0 == null) ? (VisualBasicCompilation._Closure$__.$I257-0 = new Func<MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo, DiagnosticInfo>(VisualBasicCompilation._Closure$__.$I._Lambda$__257-0)) : VisualBasicCompilation._Closure$__.$I257-0);
}
else
{
	delayedErrorInfo = ((MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo t) => ErrorFactory.ErrorInfo(ERRID.ERR_ValueTupleResolutionAmbiguous3, new object[]
	{
		t,
		conflicts.Item1,
		conflicts.Item2
	}));
}
result = new MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo(this.Assembly.Modules.get_Item(0), ref emittedName, delayedErrorInfo);
```

With a local:
```
bool flag6 = conflicts.Item1 == null;
Func<MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo, DiagnosticInfo> delayedErrorInfo;
if (flag6)
{
	Debug.Assert(conflicts.Item2 == null);
	delayedErrorInfo = ((VisualBasicCompilation._Closure$__.$I257-0 == null) ? (VisualBasicCompilation._Closure$__.$I257-0 = new Func<MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo, DiagnosticInfo>(VisualBasicCompilation._Closure$__.$I._Lambda$__257-0)) : VisualBasicCompilation._Closure$__.$I257-0);
}
else
{
	ValueTuple<AssemblySymbol, AssemblySymbol> capturedConflicts = conflicts;
	delayedErrorInfo = ((MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo t) => ErrorFactory.ErrorInfo(ERRID.ERR_ValueTupleResolutionAmbiguous3, new object[]
	{
		t,
		capturedConflicts.Item1,
		capturedConflicts.Item2
	}));
}
result = new MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo(this.Assembly.Modules.get_Item(0), ref emittedName, delayedErrorInfo);
```

Reply:
From discussion with Aleksey, I now understand the relevant difference: the closure object is only created inside the `Else` branch, when using a local within the `Else` branch. Without such local, the closure is created in the main path...

Pushed a commit with the change. Thanks